### PR TITLE
[sparkle] - enh: `RadioGroup`

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR enhances the `RadioGroup` component in the sparkle design system, aligning it with recent UI improvements and standardization efforts. The changes streamline the component's implementation while maintaining its existing functionality and introducing new features.

Key changes include:
- Refactoring the `RadioGroup` component to use `@radix-ui/react-radio-group` for improved accessibility and consistency
- Adding new variants: `RadioGroupChoice` and `RadioGroupItem` for enhanced usability
- Implementing tooltip support for radio items
- Updating the component's styling to match the new design system, with a focus on maintaining visual consistency across different states (selected, unselected, disabled)
- Removing the old `RadioButton` component in favor of the new `RadioGroup` implementation

## Risk

Broken UI

## Deploy Plan

- Publish release candidate
- Make changes in `front`
- Publish new `sparkle` version
- Deploy `front` with new `sparkle`

### CODE REVIEW

File: sparkle/src/components/Popover.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ const PopoverContent = React.forwardRef<
           "data-[side=left]:s-slide-in-from-right-2",
           "data-[side=right]:s-slide-in-from-left-2",
           "data-[side=top]:s-slide-in-from-bottom-2",
-          "s-border-border s-text-primary-950 s-z-50 s-rounded-lg s-border s-bg-background s-shadow-md s-outline-none",
+          "s-z-50 s-rounded-lg s-border s-border-border s-bg-background s-text-primary-950 s-shadow-md s-outline-none",
```
Comment (flvndvd) [inline sparkle/src/components/Popover.tsx:41]: Does `border-border` really exist?

File: sparkle/src/stories/RadioGroup.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        <div className="s-flex s-items-center s-space-x-2">
+          <RadioGroupItem value="option-three" id="option-three" size="sm" />
+          <Label htmlFor="option-two">Option Three</Label>
+        </div>
+      </RadioGroup>
+    </div>
+  );
+};
+
+export const RadioGroupWithChildrenExample = () => {
```
Comment (flvndvd) [inline sparkle/src/stories/RadioGroup.stories.tsx:67]: Is this the "assistant builder" example?

**Review by flvndvd — APPROVED** [review]
LGTM

File: sparkle/src/stories/RadioGroup.stories.tsx (hunk 2 above)
Comment (JulesBelveze) [reply sparkle/src/stories/RadioGroup.stories.tsx:67]: A simpler version but I'll rework it to be even closer to the "assistant builder" use case and basically expand the selected choice

File: sparkle/src/components/Popover.tsx (hunk 1 above)
Comment (JulesBelveze) [reply sparkle/src/components/Popover.tsx:41]: Not to my humble knowledge 😇

**Pull request merged**